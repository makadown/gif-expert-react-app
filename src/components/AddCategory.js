import React, { useState } from 'react'
import PropTypes from 'prop-types';

/**
 * Componente para capturar una nueva categoría
 * @param {agregarPapu} {setCategories} Referencia a Método del useState para agregar elemento a arreglo
 * @returns componente AddCategory
 */
export const AddCategory = ( { agregarPapu } ) => {
    const [inputValue, setInputValue] = useState('');
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2) {
            // aqui se agrega elemento al arreglo que esta en 
            // componente GifExpertApp mediante el metodo del useState.
            agregarPapu( cats => [inputValue, ...cats, ]);
            setInputValue('');
        }
    };

    return (
        <>
            <form onSubmit={ handleSubmit }>
                <input
                    type="text"
                    value={inputValue}
                    onChange={handleInputChange}
                />
            </form>
        </>
    )
}

// Candado para que se force a usar este parametro al usar este componente
AddCategory.propTypes = {
    agregarPapu: PropTypes.func.isRequired
}