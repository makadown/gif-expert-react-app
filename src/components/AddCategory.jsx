import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ( { onAddCategory } ) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) => {
        setInputValue(target.value);
    }

    const onSubmit = (e) => {
        e.preventDefault();
        const cleanInput = inputValue.trim();
        
        if (cleanInput.length <= 1) return;        
        
        if ( onAddCategory( cleanInput ) ) {
            setInputValue('');
        } else {
            window.alert('Category already exists');
        }
    }
    

    return (
        <form onSubmit={ onSubmit }>
            <input type="text" placeholder="Search gifs" id="myCategory" 
                    name="myCategory" style={{width: '25%', height: '10px'}}
                    value={inputValue} onChange={ onInputChange }
                    />
        </form>
    )
}

AddCategory.propTypes = {
    onAddCategory: PropTypes.func.isRequired,
}
