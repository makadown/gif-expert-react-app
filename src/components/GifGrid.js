import React from 'react'

export const GifGrid = ({ category }) => {

    const getGifs = async() => {
        const palabraClave = 'Rick and morty'
                .replace(' ','+').replace(' ','+').replace(' ','+');
        const apiKey = 'api_key=3Ie0XE3iXNg8qBqu6WJO2J38EheOg33Z';
        const url = 'https://api.giphy.com/v1/gifs/search?q=' +
                palabraClave + '&limit=10&'+
                apiKey;
        const respuesta = await fetch(url);
        const { data } = await respuesta.json();
        console.log(data);
        const gifs = data.map( d => {
            return { 
                id: d.id,
                title: d.title,
                url: d.images?.downsized_medium.url
            };
        });
        console.log(gifs);
    };
    getGifs();
    return (
        <>
            <li> {category} </li>
        </>
    )
}
