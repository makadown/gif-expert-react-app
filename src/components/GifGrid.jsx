import { useEffect } from "react";
import { getGifs } from "../helpers/getGifs";

export const GifGrid = ( { category } ) => {
    
    // using useEffect I need to call the getGifs function
    // - Only once and not every time the component is rendered
    // - Only if the category changes
    useEffect( () => {
        getGifs( category );
    }, [ category ] );

    

    return (
        <>
            <h3>{category} </h3>
            <p>Hola mundo!</p>
        </>
    )
}