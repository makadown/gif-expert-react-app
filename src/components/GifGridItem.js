import React from 'react'

export const GifGridItem = ( img ) => {
    console.log('recibiendo en GifGridItem', img);
    return (
        <div>
            <img src={img.url} alt={ img.title }/>
            <p>{img.title}</p>
        </div>
    );
}
