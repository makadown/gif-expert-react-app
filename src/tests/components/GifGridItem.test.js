import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas de GifGridItem', () => {
    const img = {
        id:"4N5vB4aErlVtVsywBw",
        url:"https://giphy.com/gifs/jojo-bizzare-cruisaders-4N5vB4aErlVtVsywBw",
        title:"bizzare GIF"
    };
    
    let wrapper = shallow(<GifGridItem key={img.id} {...img}/>);

    test('Debe mostrar <GifGridItem/> correctamente correctamenteu', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe tener un pÃ¡rrafo con title ', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe(img.title);
    });
    
    test('Debe tener la imagen igual al url y alt de la img', () => {
        const imagen = wrapper.find('img');
        expect(imagen.prop('src')).toBe(img.url);
        expect(imagen.prop('alt')).toBe(img.title);
    });

    test('Debe tener la class animate__fadeIn', () => {
        expect(wrapper.hasClass('animate__fadeIn'));
    })
    
})
