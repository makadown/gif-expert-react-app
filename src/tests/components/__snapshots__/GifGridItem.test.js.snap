// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifGridItem Debe mostrar <GifGridItem/> correctamente correctamenteu 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="bizzare GIF"
    src="https://giphy.com/gifs/jojo-bizzare-cruisaders-4N5vB4aErlVtVsywBw"
  />
  <p>
    bizzare GIF
  </p>
</div>
`;
